aws_signing_helper/pkcs11_signer.go

```diff
@@ -284,44 +560,83 @@ func getManufacturerId(module *pkcs11.Ctx) (string, error) {
 	return info.ManufacturerID, nil
 }
 
+// Because of *course* we have to do this for ourselves.
+//
+// Create the DER-encoded SEQUENCE containing R and S:
+//
+//	Ecdsa-Sig-Value ::= SEQUENCE {
+//	  r                   INTEGER,
+//	  s                   INTEGER
+//	}
+//
+// This is defined in RFC3279 §2.2.3 as well as SEC.1.
+// I can't find anything which mandates DER but I've seen
+// OpenSSL refusing to verify it with indeterminate length.
+func encode_ecdsa_sig_value(signature []byte) (out []byte, err error) {
+	siglen := len(signature) / 2
+
+	return asn1.Marshal(struct {R *big.Int; S *big.Int}{
```

Comment:
Potentially a stupid question, but do we have to worry about endianness here?

Reply:
An ASN.1 INTEGER is explicitly big-endian so that's fine. It's `bytesToUint` where we have to worry about endianness...